test/generate_tests.py

```diff
@@ -156,3 +156,69 @@ def test_main_drizzle(self):
 			os.remove(test_cosmo_folder+'npy_files/img_%d.npy'%(i))
 		os.rmdir(test_cosmo_folder+'npy_files')
 		os.rmdir(output_folder)
+
+	def test_h5(self):
+		# Test that the h5 kwarg generates the required h5 files
+		old_sys = copy.deepcopy(sys.argv)
+		n_generate = 21
+		output_folder = 'test_data/test_dataset'
+		sys.argv = ['test','test_data/config_dict.py',output_folder,
+			'--n',str(n_generate),'--h5']
+		if tensorflow_installed:
+			sys.argv.append('--tf_record')
+		generate.main()
+
+		image_file_list = glob.glob(os.path.join(output_folder,'image_data.h5'))
+		# Check only 1 h5 file is made:
+		self.assertEqual(len(image_file_list),1)
+		with h5py.File(image_file_list[0],'r') as f0:
+			image_data = f0['data'][()]
+		image_shape = image_data.shape
+		print('image shape',image_data.shape)
+		self.assertEqual(image_shape[0],n_generate)
+
+		# Make sure all of the files are readable and have the correct size
+		# for the config
+		for n in range(n_generate):
+			img = image_data[n,:,:]
+			self.assertTupleEqual(img.shape,(64,64))
+		os.remove(image_file_list[0])
+
+		# Make sure the metadata makes sense
```

Comment:
This block seems copied verbatim from the above tests, so could be factored out into a function. But we can also fix that later, this is  not the only place in our test suite with some duplication..